### PR TITLE
feat: comply with cmo directions using existing comply event (FPLA-866)

### PR DESCRIPTION
**Pull request by swalker125**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/FPLA-866

### Change description ###

Check on servedCaseManagementOrder size to select which directions should be complied with in the comply event.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java [hunk 1]
```diff
@@ -1277,6 +1279,50 @@ private CaseData prepareCaseData(UUID directionId,
...
+            CaseData caseData = caseDataWithSdo(buildDirections(CAFCASS))
+                .servedCaseManagementOrders(servedCaseManagementOrder(cmoDirections))
+                .build();
+
+            List<Element<Direction>> directions = service.getDirectionsToComplyWith(caseData);
+
+            assertThat(directions).isEqualTo(cmoDirections);
+        }
+
+        private CaseData.CaseDataBuilder caseDataWithSdo(List<Element<Direction>> sdoDirections) {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java]: `buildCaseDataWithSDO`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java [hunk 2]
```diff
@@ -1277,6 +1279,50 @@ private CaseData prepareCaseData(UUID directionId,
...
+            return CaseData.builder()
+                .standardDirectionOrder(
+                    Order.builder()
+                        .directions(sdoDirections)
+                        .build());
+        }
+
+        private List<Element<CaseManagementOrder>> servedCaseManagementOrder(List<Element<Direction>> cmoDirections) {
```
Comment (Jamesnelson117) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java]: `buildServedCMO`

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java (hunk 1 above)
Comment (swalker125) [reply service/src/test/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperServiceTest.java]: hmm so techincally it doesn't build as it returns a builder... I am probably not going to change to caseDataBuilderWithSdo. The method signature should give you enough information

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java [hunk 3]
```diff
@@ -70,37 +71,76 @@ void aboutToStartCallbackShouldAddAllPartiesDirectionsIntoSeparateRoleCollection
...
                 .build())
             .build();
 
         CaseData caseData = makeRequest(request, "about-to-submit");
 
-        assertThat(caseData.getStandardDirectionOrder().getDirections().get(0).getValue().getResponses()).isNotEmpty();
+        assertThat(getResponses(caseData.getServedCaseManagementOrders().get(0).getValue())).isNotEmpty();
+    }
+
+    private List<Element<DirectionResponse>> getResponses(CaseManagementOrder order) {
```
Comment (Jamesnelson117) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: `getDirectionResponses`

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java [hunk 4]
```diff
@@ -70,37 +71,76 @@ void aboutToStartCallbackShouldAddAllPartiesDirectionsIntoSeparateRoleCollection
...
+        return ImmutableList.of(Element.<CaseManagementOrder>builder()
+            .value(CaseManagementOrder.builder()
+                .directions(directions(uuid))
+                .build())
+            .build());
+    }
+
+    private Order order(UUID uuid) {
+        return Order.builder().directions(ImmutableList.of(Element.<Direction>builder()
+                .id(uuid)
```
Comment (Jamesnelson117) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: Do we need to pass / set the id prop at all? Doesn't look like you check it

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java [hunk 5]
```diff
@@ -70,37 +71,76 @@ void aboutToStartCallbackShouldAddAllPartiesDirectionsIntoSeparateRoleCollection
...
+
+    private List<Element<CaseManagementOrder>> caseManagementOrders(UUID uuid) {
+        return ImmutableList.of(Element.<CaseManagementOrder>builder()
+            .value(CaseManagementOrder.builder()
+                .directions(directions(uuid))
+                .build())
+            .build());
+    }
+
+    private Order order(UUID uuid) {
```
Comment (Jamesnelson117) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: `buildOrder`

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java [hunk 6]
```diff
@@ -70,37 +71,76 @@ void aboutToStartCallbackShouldAddAllPartiesDirectionsIntoSeparateRoleCollection
...
+                .value(Direction.builder()
+                    .directionType("example direction")
+                    .build())
+                .build()))
+                .build();
+    }
+
+    private List<Element<Direction>> directions(UUID uuid) {
+        return ImmutableList.of(Element.<Direction>builder()
+                .id(uuid)
```
Comment (Jamesnelson117) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: Again doesn't look like we check ID so not necessary needed

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java [hunk 7]
```diff
@@ -70,37 +71,76 @@ void aboutToStartCallbackShouldAddAllPartiesDirectionsIntoSeparateRoleCollection
...
+        return Order.builder().directions(ImmutableList.of(Element.<Direction>builder()
+                .id(uuid)
+                .value(Direction.builder()
+                    .directionType("example direction")
+                    .build())
+                .build()))
+                .build();
+    }
+
+    private List<Element<Direction>> directions(UUID uuid) {
```
Comment (Jamesnelson117) [inline service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: `buildDirections`

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java [hunk 8]
```diff
@@ -66,11 +67,16 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody Cal
...
-            responses, caseData.getStandardDirectionOrder().getDirections());
+        directionHelperService.addResponsesToDirections(responses, directionsToComplyWith);
 
-        caseDetails.getData().put("standardDirectionOrder", caseData.getStandardDirectionOrder());
+        //TODO: new service for sdo vs cmo in placing directions
+        if (caseData.getServedCaseManagementOrders().isEmpty()) {
+            caseDetails.getData().put("standardDirectionOrder", caseData.getStandardDirectionOrder());
+        } else {
+            caseDetails.getData().put("servedCaseManagementOrders", caseData.getServedCaseManagementOrders());
+        }
```
Comment (Jamesnelson117) [inline service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java]: Is it worth extracting this out even for testing sake?

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java (hunk 3 above)
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: variable I am getting is responses

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java (hunk 4 above)
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: directions update by UUID

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java (hunk 5 above)
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: order is fine

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java (hunk 6 above)
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: directions update by UUID

File: service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java (hunk 7 above)
Comment (swalker125) [reply service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsControllerTest.java]: directions is fine

File: service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java (hunk 8 above)
Comment (swalker125) [reply service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ComplyWithDirectionsController.java]: added the TODO and an integration test. I will move this at some point

**Pull request merged**